### ISSUE TITLE
Support UEFI Executable generation with systemd stubs

### ISSUE THREAD

**Issue by Foxboron** [anatol/booster#79]
This is probably a nice weekend hack for someone, but generating UEFI executables is a must have feature for initramfs generators these days. My WIP patch for `mkinitcpio` serves as a very nice outline.

Someone that want to try contribute to this project can probably take a stab at it :) Else I can implement it once I have less on my todo list. I'll help review it if needed!

https://github.com/archlinux/mkinitcpio/pull/53

**Comment by anatol** [comment]
Yes the idea is interesting.

Looking at your patch - how do you sign the result efi boot image?

**Comment by Foxboron** [comment]
dracut allows you to specify a signing key and it uses `sbsign`. For `mkinitcpio` I'd rather implement some "post image generation" hook that would let you use [sbctl](https://github.com/Foxboron/sbctl/), but personally this is run with a separate pacman hook for `sbctl` currently.

**Comment by Foxboron** [comment]
If you do want to use golang for this, you could consider using `go-uefi`.

https://github.com/Foxboron/go-uefi/blob/master/cmd/gosign/main.go

But the API isn't really stable and needs a lot more work.

**Comment by anatol** [comment]
Yeah, a golang lib is preferable. API stability is not much of concern (thanks to golang modules).

**Comment by c3Ls1US** [comment]
Same topic as https://github.com/anatol/booster/issues/110. It doesn't necessarily need to be in Booster.

**Comment by thePrivacyFanatic** [comment]
since the code is distro specific maybe [booster uki manager](https://github.com/Zile995/booster-um) would solve it better?
